metadata.mk

```diff
@@ -3,7 +3,7 @@
 #################################################################################################
 
 # The version of github.com/projectcalico/go-build to use.
-GO_BUILD_VER = v0.85
+GO_BUILD_VER = llvm-15
```

Comment:
- [ ] Use official go-build version once https://github.com/projectcalico/go-build/pull/470 is reviewed/merged.